### ISSUE TITLE
Chat - Download option in three-dot menu is still available in offline mode

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#36901]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.43-2
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Go to any chat
3. Upload a video
4. Go offline
5. Tap on the video
6. Note that the download option on the top right disappears
7. Tap 3-dot menu

## Expected Result:
The download option will not be available

## Actual Result:
The download option is still available

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/a82c142d-7388-490a-a0d3-b1f3c4b53122

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp
CC @quinthar

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The video download option is still available while offline.

### What is the root cause of that problem?
We always include the download option.
https://github.com/Expensify/App/blob/669ac330b4eeaad2238419bdb501811d3e031e13/src/components/VideoPlayerContexts/VideoPopoverMenuContext.js#L33-L41

### What changes do you think we should make in order to solve the problem?

Use `useNetwork` hook to check network state and don't include the download option if `isOffline` is true.

**Comment by bondydaa** [comment]
pretty sure this PR is the cause of the regression https://github.com/Expensify/App/pull/30829, though I guess "regression" might not be the correct terminology here since that PR was introducing a new feature.

I think we can just create a new PR, assigning @Krishna2323

**Comment by bondydaa** [comment]
I'm also going to remove the deploy blocker here, I don't think it's worth blocking the deploy over this since it's a minor thing IMO. if someone disagrees feel free to re-add that label

**Comment by bondydaa** [comment]
@twisterdotcom you can ignore for now, added you just to help manage payment once the PR is done.

**Comment by abdulrahuman5196** [comment]
Got it. Do let me know once PR is available. I will take a look into it.

**Comment by abdulrahuman5196** [comment]
>The PR that introduced the bug has been identified. Link to the PR:
The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Not a regression. Seems to be the case for sometime.

>Determine if we should create a regression test for this bug.

Yes.

>If we decide to create a regression test for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

1) Navigate to staging.new.expensify.com
2) Go to any chat
3) Upload a video
4) Go offline
5) Tap on the video
6) Note that the download option on the top right disappears
7) Tap 3-dot menu
8) Verify download option is not shown

@twisterdotcom Added BZ checklist. Seems melvin didn't auto update here. This issue is already due for payment.

**Comment by abdulrahuman5196** [comment]
@twisterdotcom . Accepted the offer

**Comment by twisterdotcom** [comment]
Payments released.
